File: angular/src/app/models/message.ts
export class Message {
  content: string;
  date: Date;
  from: string;
  to: string;
  queueName: string;

  constructor(
    from: string,
    to: string,
    content: string,
    date: Date = new Date(),
    queueName: string
  ) {
    this.from = from;
    this.to = to;
    this.content = content;
    this.date = date;
    this.queueName = queueName;
  }
}

File: src/main/java/com/wittano/websocket/models/entity/Message.java
package com.wittano.websocket.models.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import java.util.Date;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Message {
    @Id
    private long id;
    @ManyToOne(cascade = {CascadeType.PERSIST, CascadeType.DETACH})
    private User from;
    @ManyToOne(cascade = {CascadeType.PERSIST, CascadeType.DETACH})
    private User to;
    private String content;
    private Date date = new Date();
    private String queueName;
}
